engine/crates/runtime/src/fetch.rs

```diff
@@ -60,6 +72,58 @@ impl Fetcher {
             inner: Arc::new(fetcher),
         }
     }
+
+    #[allow(unused_variables)]
+    pub async fn post<SleepFut>(
+        &self,
+        request: &FetchRequest<'_>,
+        sleep: impl Fn(Duration) -> SleepFut + Send + Sync,
```

Comment:
`sleep` shouldn't be part of the trait, the implementation can do sleep by itself without any problem. It's aware of Cloudflare/Tokio. I's only the engine that can't rely on tokio. `runtime-local` can

Reply:
Yeah but the implementation of retries needs to move from runtime to runtime local.

Reply:
Honestly I would move all of this to `engine-v2`, would make more sense as you rely on retry budget, rate limiter and sleep

Reply:
will remove `subgraph_name` from the `FetchRequest` also, shouldn't be there, it was an error. `FetchError` should just have a timeout case